### PR TITLE
Crayons have powder back, but it doesn't color anymore

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

This PR readds the individual types of coloured powder to the crayons, but it also makes them unable to color you via eating them.

## Why It's Good For The Game

There's a few food recipes I found that got screwed over by the crayon nerf, since there's no way to get powder outside of adminbus as-is. To be able to craft them again, I made this PR while making sure to keep in the reason why it was removed to begin with.

## Changelog
:cl:
tweak: crayons once again have powder but it doesnt recolor, for your green burger crafting needs
/:cl:

### CODE REVIEW

**Comment by kriskog** [comment]
You can currently get the powder by growing rainbow flowers in botany. The intention when the powders were removed from the crayons were to leave them (and the recoloring) available with a bit of work instead of just going to the closest crayon box to get some.

**Comment by carlarctg** [comment]
guess i'll give crayon a subtype then or something

**Comment by carlarctg** [comment]
> You can currently get the powder by growing rainbow flowers in botany. The intention when the powders were removed from the crayons were to leave them (and the recoloring) available with a bit of work instead of just going to the closest crayon box to get some.
@kriskog think i fixed it

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1289,6 +1287,38 @@
 	color = "#FFFFFF" // white
 	random_color_list = list("#FFFFFF") //doesn't actually change appearance at all
 
+ /* used by crayons, can't color living things but still used for stuff like food recipes */
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: what the frick is this

Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: coby you can't just say that without context i am very dumb and do not understand what you are complaining about

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: everything below this

Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: it's the same as color powder but the procs for colouring mobs return, so you can't use it to turn red or black or whatever by itself

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: did you test this? pretty sure you get icky side effects with just flat out returning life and reaction.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1406,15 +1440,18 @@
 	color = "#C8A5DC"
 	var/list/random_color_list = list("#00aedb","#a200ff","#f47835","#d41243","#d11141","#00b159","#00aedb","#f37735","#ffc425","#008744","#0057e7","#d62d20","#ffa700")
 	taste_description = "rainbows"
+	var/can_colour_mobs = TRUE
 
 
 /datum/reagent/colorful_reagent/on_mob_life(mob/living/carbon/M)
-	M.add_atom_colour(pick(random_color_list), WASHABLE_COLOUR_PRIORITY)
-	..()
+	if(can_colour_mobs == TRUE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	if(can_colour_mobs)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -1406,15 +1440,18 @@
...
+		M.add_atom_colour(pick(random_color_list), WASHABLE_COLOUR_PRIORITY)
+		..()
 
 /datum/reagent/colorful_reagent/reaction_mob(mob/living/M, reac_volume)
-	M.add_atom_colour(pick(random_color_list), WASHABLE_COLOUR_PRIORITY)
-	..()
+	if(can_colour_mobs == TRUE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	if(can_colour_mobs)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -1406,15 +1440,18 @@
...
 	taste_description = "rainbows"
+	var/can_colour_mobs = TRUE
 
 
 /datum/reagent/colorful_reagent/on_mob_life(mob/living/carbon/M)
-	M.add_atom_colour(pick(random_color_list), WASHABLE_COLOUR_PRIORITY)
-	..()
+	if(can_colour_mobs == TRUE)
+		M.add_atom_colour(pick(random_color_list), WASHABLE_COLOUR_PRIORITY)
+		..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
		return ..()
```
we have to return TRUE for it to metabolize

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: you could also just set `. = ..()`

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: fairly sure i tested and worked, with drinking it not working but throwing it at tiles painting them will check again. what icky side effects?

**Pull request merged**